### PR TITLE
Toggle details with redux

### PR DESCRIPTION
**Pull request by ka1130**
This PR adds the possibility to toggle details (summary) of a show and pagination that displays maximum of 12 shows per page. 

Toggling details is handled in two ways:
- via Redux => displaying modal with further info of a clicked show
- via HTML5 `<details>` tag that is widely supported by modern browsers and offers the possibility to toglle open/close `<summary>` with no JS whatsoever

Pagination is implemented by hand, using Redux.

The maximum page's width is set to 80 rem and Sass variables are introduced for better control of the app's color palette.

**Disclaimer**: _There's tons of comments in the code, it's just temporary and helps me understand the Redux flow. Hope they may stay until I get more comfortable with Redux. They will definitely be gone at further stages of work on the project._

**Screencast**: 
peek 2018-10-07 17-28

### CODE REVIEW

File: src/components/App/ActiveShowModal/index.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { hideDetails } from 'redux/actions/detailsActions';
+
+import imgMissing from 'images/img_missing.png';
+
+import styles from './ActiveShowModal.module.scss';
+
+class ActiveShowModal extends Component {
```
Comment (yakato) [inline src/components/App/ActiveShowModal/index.js]: I like the idea of modal with details, but as you wrote - it allows to preview details only for one episode at the time. Which is also possible with pure React solution. I still think that you should stay with pure React but do a modal style details when user click on the poster. And add new button (except the one in the modal) to go to the Show.

File: src/components/App/ActiveShowModal/index.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  
+  render() {
+    const { isVisible, activeShow } = this.props;
+    return (
+      <div className={`${isVisible ? styles.visible : ''} ${styles.modalWrapper}`}>
+        <div className={styles.modalContent}>
+          <button
+            className={styles.close}
+            onClick={() => this.props.hideDetails()}
```
Comment (yakato) [inline src/components/App/ActiveShowModal/index.js]: It should be possible to close modal also when click outside.

File: src/components/App/ActiveShowModal/index.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          >
+            &times;
+          </button>
+          <figure>
+            <img
+              src={activeShow && activeShow.activeShow.show.image.medium ? activeShow.activeShow.show.image.medium : imgMissing}
```
Comment (yakato) [inline src/components/App/ActiveShowModal/index.js]: Maybe extract this long prop into `const` above?

File: src/components/App/ActiveShowModal/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+              alt="" />
+            <figcaption>
+              <h5 className={styles.showTitle}>
+                {activeShow ? activeShow.activeShow.name : ''}
```
Comment (yakato) [inline src/components/App/ActiveShowModal/index.js]: If there is a lot of conditions for `activeShow` maybe consider another function with jsx for activeShow and render it once conditionally?

File: src/components/App/EpisodesList/Show/index.js [hunk 5]
```diff
@@ -1,42 +1,49 @@
...
-    return {
-      name: nextProps.name,
-      imgUrl: saveImgUrl,
-    }
-  }
-
   render() {
-    const { name, imgUrl } = this.state;
+    const { name, imgUrl, show } = this.props;
+    const summary = show.summary
```
Comment (yakato) [inline src/components/App/EpisodesList/Show/index.js]: Isn't it doubled in another file?

File: src/components/App/EpisodesList/Show/index.js [hunk 6]
```diff
@@ -1,42 +1,49 @@
...
+                      // strip the summary from HTML tags parsed as plain text
+                      : 'No summary available';
     return (
       <li className={styles.showListElement}>
         <figure>
           <img src={imgUrl} alt={name} />
           <figcaption>
-            <h5>{name}</h5>
+            <details>
+              <summary>{name}</summary>
```
Comment (yakato) [inline src/components/App/EpisodesList/Show/index.js]: This `summary` tag shouldn't be for actual summary?

File: src/components/App/index.js [hunk 7]
```diff
@@ -1,39 +1,63 @@
...
   componentDidMount() {
-    this.props.dispatch(fetchEpisodes());
+    this.props.fetchEpisodes();
   }
 
   render() {
-    const { error, loading, episodes } = this.props;
+    const { error, loading, episodes, activeShow } = this.props;
+    const { currentPage } = this.props.currentPage;
+    const episodesPerPage = 12; //arbitrary number, fits nicely into various screen widths
```
Comment (yakato) [inline src/components/App/index.js]: If the number of episodes per page fits almost always why not hardcode it inside pagination? Or change dynamically according to screen size?

File: src/components/App/index.js [hunk 8]
```diff
@@ -1,39 +1,63 @@
...
     );
   }
 }
 
+const mapDispatchToProps = dispatch => (
+  bindActionCreators({ fetchEpisodes }, dispatch)
+);
+
 const mapStateToProps = state => ({
+  activeShow: state.activeShow,
```
Comment (yakato) [inline src/components/App/index.js]: name is unfortunate - the Episode is active not the Show right?

File: src/redux/actions/types.js [hunk 9]
```diff
@@ -1,3 +1,8 @@
 export const FETCH_EPISODES_BEGIN   = 'FETCH_EPISODES_BEGIN';
 export const FETCH_EPISODES_SUCCESS = 'FETCH_EPISODES_SUCCESS';
 export const FETCH_EPISODES_FAILURE = 'FETCH_EPISODES_FAILURE';
+
+export const SHOW_SELECTED = 'SHOW_SELECTED';
```
Comment (yakato) [inline src/redux/actions/types.js]: I'd suggest to pay more attention about naming. If I will be new in this project what the 'SHOW_SELECTED' action tells me about what it does?

**Comment by ka1130** [comment]
Closing because [PR#9](https://github.com/ka1130/TVloVer/pull/9) delivers the same features, refactored.

**Pull request closed without merging**